src/main/java/logic/Storage.java

```diff
@@ -0,0 +1,118 @@
+package logic;
+
+import task.*;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+import java.util.regex.Pattern;
+
+/**
+ * Deals with loading tasks and saving tasks in file
+ */
+
+public class Storage {
+    private String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Reads and scans text file, convert them to Task Objs to be added
+     *
+     * @return List of Tasks loaded from text file
+     * @throws DukeException If encounter file creation/parsing problems
+     */
+    public List<Task> load() throws DukeException {
+        List<Task> taskList = new ArrayList<>();
+
+        try {
+            File f = new File(filePath); //hardCoded file directory
+            Scanner sc = new Scanner(f);
+
+            while (sc.hasNext()) {
+                String line = sc.nextLine();
+                String[] strArr = line.split(Pattern.quote(" | "));
+                Task task = readAndCreateTask(strArr);
+                taskList.add(task);
+            }
+
+            Ui.printStr("Data is loaded from data/taskList.txt");
+        } catch (FileNotFoundException e) {
+            Ui.printStr("File not found, data not loaded!\nCreating new file taskList.txt");
+            File f = new File(filePath);
+
+            try {
+                f.createNewFile(); //creates new text file
+            } catch (IOException ioE) {
```

Comment:
Might help to name the variable `e` as a standard way to read exceptions.

Reply:
Did not name e due to the variable e in the parent scope